test/distributed/test_ddp_under_dist_autograd.py

```diff
@@ -545,6 +546,51 @@ def test_ddp_dist_autograd_sparse_grads(self):
             self.assertEqual(1, len(grads_dict))
             self.assertEqual(model.weight.grad, grads_dict[model.weight])
 
+    @staticmethod
+    def get_remote_grads(rref, context_id):
+        return dist_autograd.get_gradients(context_id)[rref.local_value().weight]
+
+    @requires_gloo()
+    @dist_init
+    def test_ddp_dist_autograd_local_vs_remote(self):
+        # Each trainer uses a different random seed. Otherwise, they are going
+        # to have exactly the same initial model parameters, input, and
+        # therefore grads. That means the grads will be the same before and
+        # after DDP's all-reduce.
+        torch.manual_seed(self.rank)
+        dist.init_process_group(
+            backend="gloo",
+            init_method="file://{}".format(self.file_name),
+            world_size=self.world_size,
+            rank=self.rank)
+
+        remote_layer1 = RemoteModule("worker0", nn.Linear, args=(10, 5, False))
+        layer1 = nn.Linear(10, 5, False)
+        # Start with the same parameters for remote and local
+        layer1.weight = remote_layer1.module_rref.to_here().weight
+
+        # Run local case.
+        layer2 = nn.Linear(5, 1)
+        inputs = torch.rand((10, 10))
+        ddp_model = DistributedDataParallel(layer2)
```

Comment:
Any reason why the ddp_model doesn't wrap the layer1/remote layer for both cases? 
i.e. `local_ddp_model = DDP(nn.Sequential(layer1, layer2))`, and `ddp_model_with_remote = DDP(nn.Sequential(remote_layer_1, layer2))`? Should the test still work as expected in this case?

Reply:
In this test, I didn't intend to replicate layer1 via DDP thats why its separate. Also, if we put remote_layer1 as part of DDP, it won't be replicated whereas the local layer1 would be replicated. So that might cause some discrepancy in the results.